src/aiod_sdk/calls/calls.py

```diff
@@ -0,0 +1,165 @@
+):
+    """
+    Asynchronously retrieve metadata for a list of ASSET_TYPE identifiers.
+
+    Parameters:
+        identifiers (list[int]): The list of identifiers of the ASSET_TYPE to retrieve.
+        version (str | None): The version of the endpoint (default is None).
+        format (Literal["pandas", "dict"]): The desired format for the response (default is "pandas").
+
+    Returns:
+        pd.Series | dict: The retrieved metadata for the specified ASSET_TYPE.
+    """
+    urls = [
+        url_to_get_asset(asset_type, identifier, version) for identifier in identifiers
+    ]
+    response_data = await _fetch_resources(urls)
+    resources = format_response(response_data, data_format)
+    return resources
+
+
+async def get_list_async(
+    asset_type: str,
+    offset: int = 0,
+    limit: int = 100,
+    batch_size: int = 10,
+    version: str | None = None,
+    data_format: Literal["pandas", "dict"] = "pandas",
+) -> pd.DataFrame | dict:
+    """
+    Asynchronously retrieve a list of ASSET_TYPE from the catalogue in batches.
+
+    Parameters:
+        offset (int): The offset for pagination (default is 0).
+        limit (int): The maximum number of items to retrieve (default is 10).
+        batch_size (int): The number of items in a a batch.
+        version (str | None): The version of the endpoint (default is None).
+        data_format (Literal["pandas", "dict"]): The desired format for the response (default is "pandas").
+
+    Returns:
+        pd.DataFrame | dict: The retrieved metadata in the specified format.
+    """
+    if batch_size <= 0:
+        raise ValueError(
+            "batch_size must be larger than 0, otherwise you can use the synchronous get_list function!"
+        )
+    current_offset = offset
+    next_limit = min(batch_size, limit)
+    batches = []
+    while current_offset < offset + limit:
+        batches.append((current_offset, next_limit))
+        current_offset += next_limit
+        next_limit = min(batch_size, offset + limit - current_offset)
+
+    urls = [
+        url_to_get_list(asset_type, offset, limit, version) for offset, limit in batches
+    ]
+
+    response_data = await _fetch_resources(urls)
+    flattened_response_data = [
+        response for batch in response_data for response in batch
+    ]
+    resources = format_response(flattened_response_data, data_format)
+    return resources
+
+
+async def _fetch_resources(urls) -> dict:
+    async def _fetch_data(session, url) -> dict:
+        async with session.get(url) as response:
+            return await response.json()
+
+    async with aiohttp.ClientSession() as session:
+        tasks = [_fetch_data(session, url) for url in urls]
+        response_data = await asyncio.gather(*tasks)
+    return response_data
+
+
+wrap_common_calls = partial(wrap_calls, calls=[get_list, counts, get_asset])
+wrap_common_async_calls = partial(wrap_calls, calls=[get_asset_async, get_list_async])
```

Comment:
Are there some modules where we do not want to support `async` calls? And is this a common occurrence?
Otherwise I am not sure we should separate everything here.

Reply:
Indeed. There are some `GET` endpoints that won't require it. But I think they will be structured in a different way or at least in a separate folder. I'm putting this calls together.